### PR TITLE
GNMI-1.2: Adding deviation for authentication and updating ISIS configuration

### PR DESCRIPTION
**Pull request by trathod1**
Following changes are made under this PR - 
   1. Updating configuration to use existing deviation ISISprotocolEnabledNotRequired
   2. Adding subinterface to default-NI using deviation ExplicitInterfaceInDefaultVRF
   3. Introducing deviation for ISISGlobalAuthenticationNotRequired to avoid Global.Authentication-check for ISIS
   4. Introducing deviation for ISISLevelAuthenticationNotRequired to avoid Level Authentication configuration for ISIS
   5. Enabling ISIS ipv4-unicast at global config.

"This code is a Contribution to the OpenConfig Feature Profiles project ("Work") made under the Google Software Grant and Corporate Contributor License Agreement ("CLA") and governed by the Apache License 2.0. No other rights or licenses in or to any of Nokia's intellectual property are granted for any other purpose. This code is provided on an "as is" basis without any warranties of any kind."

### CODE REVIEW

File: internal/deviations/deviations.go [hunk 1]
```diff
@@ -222,4 +222,10 @@ var (
 
 	ISISRestartSuppressUnsupported = flag.Bool("deviation_isis_restart_suppress_unsupported", false,
 		"Device skip isis restart-suppress check if value is true, Default value is false")
+
+	ISISGlobalAuthenticationNotRequired = flag.Bool("deviation_isis_global_authentication_not_required", false,
```
Comment (prinikasn) [inline internal/deviations/deviations.go:226]: Please follow the new style for any new deviations. The flag should be unexported and accessed through an exported accessor method that accepts a DUT, e.g.
https://github.com/openconfig/featureprofiles/blob/main/internal/deviations/deviations.go#L73
This is going to enable our transition away from flags to reading vendor-specific deviations from the metadata textproto.

Sample PR to see the deviation and use in test - https://github.com/openconfig/featureprofiles/pull/1358/files

**Review by prinikasn — CHANGES_REQUESTED** [review]

File: internal/deviations/deviations.go (hunk 1 above)
Comment (trathod1) [reply internal/deviations/deviations.go:226]: Sure. I will make the changes accordingly and push the commit.

Comment (trathod1) [reply internal/deviations/deviations.go]: Updated deviations as per new style.

**Comment by jjing1122** [comment]
Filed internal bug https://b.corp.google.com/issues/277230687 for presubmit process

**Review by jjing1122 — DISMISSED** [review]

**Comment by jjing1122** [comment]
@trathod1 please rebase

**Comment by trathod1** [comment]
@jjing1122 Resolved conflicts and rebased it.

**Review by jjing1122 — APPROVED** [review]

**Review by prinikasn — APPROVED** [review]

**Pull request merged**